lib/validation.tsx

```diff
@@ -106,26 +108,59 @@ const isFieldResponseValid = (
       break;
     }
     case "fileInput": {
+      //TODO need refactoring.
       const typedValue = value as { file: File; src: FileReader; name: string; size: number };
-      if (validator.required && typedValue.file === null) return t("input-validation.required");
+      if ((validator.required && typedValue === undefined) || typedValue === null)
+        return t("input-validation.required");
+      if (validator.required && typedValue && typedValue.file === null)
+        return t("input-validation.required");
       // Size limit is 8 MB
-      if (typedValue.size > 8000000) return t("input-validation.file-size-too-large");
+      if (typedValue && typedValue.size > 8000000) return t("input-validation.file-size-too-large");
```

Comment:
There's a nice shorthand that exists now with ES2020 that allows for Optional Chaining.  [Reference Doc](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)

It allows for expressions like `if( typedValue && typedValue.size)` to be rewritten as `if(typedValue?.size)` which greatly reduces the overhead and cognitive overload when writing these types of boolean logic checks.